sw/animation/animation_thread.h

```diff
@@ -47,6 +51,19 @@ void main_loop_function()
       drawer.agent(ID, s[ID]->state.at(0), s[ID]->state.at(1), s[ID]->orientation);
       // Input: ID, p_x global, p_y global, v_x global, v_y global
       drawer.velocity_arrow(ID,  s[ID]->state.at(0), s[ID]->state.at(1), s[ID]->state.at(2), s[ID]->state.at(3));
+      
```

Comment:
What happens here if the sensors are different. All drawing functions that are specific to your agent should be written in the draw function of the agent and/or controller